backend/clap/src/main/java/zpi/ppea/clap/mappers/DetailedSubmissionMapper.java

```diff
@@ -0,0 +1,33 @@
+package zpi.ppea.clap.mappers;
+
+import data_store.DetailsSubmissionResponse;
+import zpi.ppea.clap.dtos.AppReportDto;
+import zpi.ppea.clap.dtos.DetailsSubmissionResponseDto;
+import zpi.ppea.clap.dtos.ProjectState;
+
+public class DetailedSubmissionMapper {
+    private DetailedSubmissionMapper() {
+    }
+
+    public static DetailsSubmissionResponseDto mapToDto(DetailsSubmissionResponse response) {
+
+        AppReportDto appReportDto = AppReportDto.builder()
+                .isDraft(response.getReport().getIsDraft())
+                .reportSubmissionDate(response.getReport().getReportSubmissionDate())
+                .projectGoals(response.getReport().getProjectGoals())
+                .organisationStructure(response.getReport().getOrganisationStructure())
+                .divisionOfWork(response.getReport().getDivisionOfWork())
+                .projectSchedule(response.getReport().getProjectSchedule())
+                .attachments(response.getReport().getAttatchments())
+                .build();
+
+        return DetailsSubmissionResponseDto.builder()
+                .teamSize(response.getTeamSize())
+                .finishDate(response.getFinishDate())
+                .status(ProjectState.valueOf(response.getStatus().name()))
+                .budget(response.getBudget())
+                .description(response.getDescription())
+                .appReportDto(appReportDto)
+                .build();
+    }
```

Comment:
Have you thought about using some mapping library? I used [https://mapstruct.org/](mapstruct) that automatically generates these, you just write interface:
```java


        @Mapper 1
        public interface CarMapper {
         
            CarMapper INSTANCE = Mappers.getMapper( CarMapper.class ); 3
         
            @Mapping(source = "numberOfSeats", target = "seatCount")
            CarDto carToCarDto(Car car); 2
        }


```

it's not necessary but for more dtos the complexity will grow, this is just one flow. It's up to you

Reply:
I tried, but for me it's easier and faster to use mapper classes.